File: java/api/src/main/java/eu/ill/cameo/api/base/Event.java
/*
 * CAMEO
 *
 * Copyright 2015 Institut Laue-Langevin
 *
 * Licensed under BSD 3-Clause and GPL-v3 as described in license files.
 * You may not use this work except in compliance with the Licences.
 *
 */

package eu.ill.cameo.api.base;

/**
 * Base class for events.
 */
public class Event {
	
	protected int id;
	protected String name;
	
	/**
	 * Constructor.
	 * @param id The application id.
	 * @param name The application name.
	 */
	public Event(int id, String name) {
		super();
		this.id = id;
		this.name = name;
	}

	/**
	 * Gets the application id.
	 * @return The application id.
	 */
	public int getId() {
		return id;
	}

	/**
	 * Gets the application name.
	 * @return The application name.
	 */	
	public String getName() {
		return name;
	}
}

File: cpp/api/src/base/Event.cpp
/*
 * CAMEO
 *
 * Copyright 2015 Institut Laue-Langevin
 *
 * Licensed under BSD 3-Clause and GPL-v3 as described in license files.
 * You may not use this work except in compliance with the Licences.
 *
 */

#include "Event.h"

namespace cameo {

Event::Event(int id, const std::string& name) :
	m_id{id},
	m_name{name} {
}

Event::Event(const Event& event) :
	m_id{event.m_id},
	m_name{event.m_name} {
}

Event::~Event() {
}

int Event::getId() const {
	return m_id;
}

const std::string& Event::getName() const {
	return m_name;
}

}